nagios.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
```

Comment:
@vchepeli the `outFor` and `errOutFor` are passed as arguments into `nagiosData` here:
https://github.com/philbrookes/fh-system-dump-tool/blob/RHMAP-8114/nagios.go#L19

The anonymous function seems sensible to me, as it is such a specific use case, I don't see what we'd gain by defining the function outside and calling it here?